### PR TITLE
Set-AzVMDiskEncryptionExtension: Improve error message for -skipVmBackup 

### PR DESCRIPTION
**Pull request by Jyotsna-Anand**
## Description

* Improved the error message displayed when -skipVmBackup parameter is not specified for Linux VMs with managed disks.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs [hunk 1]
```diff
@@ -541,9 +541,22 @@ public override void ExecuteCmdlet()
...
 
+                    var vmParameters = (this.ComputeClient.ComputeManagementClient.VirtualMachines.Get(
+                        this.ResourceGroupName, VMName));
+
                     if (OperatingSystemTypes.Linux.Equals(currentOSType) && !SkipVmBackup)
                     {
-                        CreateVMBackupForLinx();
+                        if (vmParameters.StorageProfile.OsDisk.ManagedDisk != null)
+                        {
+                            ThrowTerminatingError(new ErrorRecord(new ArgumentException(string.Format(CultureInfo.CurrentUICulture, "-skipVmBackup parameter is a required parameter for encrypting Linux VMs with managed disks. For more information, see https://docs.microsoft.com/azure/security/azure-security-disk-encryption-linux.")),
```
Comment (niander) [inline src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs]: We are trying to have all strings taken from the Resources.resx file. Could you do it?

Comment (Jyotsna-Anand) [reply src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs]: I am trying to throw an error in the powershell window and not have a pop-up window display the error. Existing usages of Resources.resx file seem to be having a pop-up displaying the message.  Should I still be using Resources.resx?

Comment (niander) [reply src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs]: Resources.resx would just hold the string for future localization

**Comment by niander** [comment]
@Jyotsna-Anand Any update on this?

File: src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs (hunk 1 above)
Comment (Jyotsna-Anand) [reply src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs]: @niander I have added changes. Can you please take a look?

**Comment by niander** [comment]
@Jyotsna-Anand Thanks for addressing my comments. Please, take a look at @markcowl review comments.

**Comment by markcowl** [comment]
@Jyotsna-Anand Do you plan to update this PR as specified?  If not, I will close at EOW.

**Comment by Jyotsna-Anand** [comment]
@Mark Cowlishaw<mailto:notifications@github.com> I will be updating this PR early next week. I got side tracked finishing up another work item this week. Thanks for following up.

**Comment by cormacpayne** [comment]
@Jyotsna-Anand I'm going to temporarily close this PR until the comments above can be addressed. Feel free to re-open the PR whenever you're able to address the comments.

**Pull request closed without merging**